test/widgets/compose_box_test.dart

```diff
@@ -31,14 +31,27 @@ void main() {
   late PerAccountStore store;
   late FakeApiConnection connection;
 
-  Future<GlobalKey<ComposeBoxController>> prepareComposeBox(WidgetTester tester,
-      {required Narrow narrow, List<User> users = const []}) async {
+  Future<GlobalKey<ComposeBoxController>> prepareComposeBox(WidgetTester tester, {
+    required Narrow narrow,
+    User? selfUser,
+    int? realmWaitingPeriodThreshold,
+    List<User> users = const [],
+    List<ZulipStream> streams = const [],
+  }) async {
+    if (narrow is ChannelNarrow || narrow is TopicNarrow) {
+      final channelId = narrow is ChannelNarrow ? narrow.streamId : (narrow as TopicNarrow).streamId;
+      assert(streams.any((stream) => stream.streamId == channelId),
+        'Add a channel with "streamId" the same as of $narrow.streamId to the store.');
+    }
     addTearDown(testBinding.reset);
-    await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+    final account = eg.account(user: selfUser ?? eg.selfUser);
```

Comment:
Because the same self-user object is needed below, best to do the `?? eg.selfUser` fallback just once, to avoid accidental mismatch on future changes to this code. So:
```suggestion
    selfUser ??= eg.selfUser;
    final account = eg.account(user: selfUser);
```